doc/source/blueprint_cce_vault.rst

```diff
@@ -0,0 +1,592 @@
+       policies=tls-zk-ro \
+       ttl=2h
+
+With this done token of the service account zookeeper in the zookeeper
+namespace would be able to access to the vault for reading secrets located
+under `/secret/tls` path. And since it is higly recommended to follow the least
+required privilege principle only read only access to the TLS data is granted.
+A time to live of 3 hours is being used here meaning that once application
+authorize to Vault the token it gets can be used during next 2 hours. This need
+to be carefully aligned with the time to live or the service account token to
+minimize their overlap. It is advised to keep it relatively short.
+
+This is one the most sensitive steps in the whole configuration, since it can
+not be generally excluded that the applications deployed in the Kubernetes find
+escape or be compromised. For that case it is required to strictly limit amount
+of secrets accessor will be able to read.
+
+Populating secrets in Vault
+===========================
+
+Within Vault there are 2 possibilities to access TLS certificates:
+
+* Store certificate data in the `KeyValue store
+  <https://www.vaultproject.io/docs/secrets/kv/kv-v2>`_
+
+* Use `PKI secrets engine <https://www.vaultproject.io/docs/secrets/pki>`_ to
+  issue certificates
+
+Vault enables users not only to store TLS certificates data in the KV store,
+but also to create and  revoke them. To keep this tutorial simple enough we are
+not going to do this and only upload generated certificates into the KV store.
+A lesson on improving that is left to the reader.
+
+.. code-block:: shell
+
+   $ vault kv put secret/tls/ca certificate=@ca.crt
+   $ vault kv put secret/tls/zk_server certificate=@zk_server.crt private_key=@zk_server.key
+   $ vault kv put secret/tls/zk_client certificate=@zk_client.crt private_key=@zk_client.key
+
+Deploying Zookeeper
+===================
+
+Now that the secrets are stored safely in Vault and only allowed applications
+can fetch them it is time to look how exactly the application accesses the
+secrets. Generally utilizing Vault requires modification of the application.
+`Vault agent <https://www.vaultproject.io/docs/agent>`_ is a tool that was
+created to simplify secrets delivery for applications where it is hard or
+impossible to change the application itself.  Agent is taking care of reading
+secrets from Vault and can deliver them to the file system.
+
+1. Create a Kubernetes namespace named `zookeeper`.
+
+.. code-block:: shell
+
+   $ kubectl create namespace zookeeper
+
+2. Create a Kubernetes service account named `zookeeper`.
+
+.. code-block:: shell
+
+   $ kubectl create serviceaccount zookeeper
+
+3. In Kubernetes a *service account* provides an identity for the services
+   running in the POD so that the process can access Kubernetes API. The same
+   identity can be used to access Vault, but require one special permission -
+   access to the tokenreview API. If a dedicated reviewer JWT is used this step
```

Comment:
you lost me on "tokenreview API" and "reviewer JWT". Please include some references or spend a sentence on those.